server/storages/table_helper.go

```diff
@@ -88,6 +90,14 @@ func (th *TableHelper) EnsureTable(destinationName string, dataSchema *adapters.
 		return dbSchema, nil
 	}
 
+	if th.max_columns > 0 {
+		columns_count := len(dbSchema.Columns) + len(diff.Columns)
+		if columns_count > th.max_columns {
+			msg := fmt.Sprintf("Count of columns %d should be less or equal max_columns %d", columns_count, th.max_columns)
```

Comment:
return `fmt.Errorf("", ...)` could be used here.

Reply:
Changed